### DISCUSSION TITLE
Thanks!

### DISCUSSION THREAD

**Discussion by F1DNA** [madMAx43v3r/chia-plotter#106] [General]
So, I was grinning ear to ear when I fired this up.  Not often I push my R520 this hard and the jet engine sound emanating from the server room has me all giddy!  When I saw 110GB ramdisk in your readme, I was like, fuckin a' right doggie!  Let's do this!

Plus this is a beautiful sight!

![image](https://user-images.githubusercontent.com/61926834/121454191-fab65580-c967-11eb-97c7-88afced47f19.png)

Specs:
R520
2x E5-2470 v2 @ 2.40GHz - 20vCPUs allocated to this VM
192GB of RAM - 122GB allocated to this VM
2x SATA SSDs - Older Samsung EVO something somethings, don't recall - in RAID 0
/dev/shm with 112GB allocated for tmp2
Final destination is an NFS share - this will be my bottleneck but unfortunately I am out of bays on this server and it's used in "production" in my homelab so nothing more I can do about that and USB 2.0 isn't a much better option.  At least it wasn't when I was looking at ~8 hour plot times, I spose now if I save 15 minutes, might be worth it.

Here are my results so far, not too far off from madMAx43v3r.  Very similar hardware specs, your CPUs are slightly better, I believe so I'm happy with these numbers:

nohup ./chia_plot -n 1 -r 16 -u 128 -t /mnt/tmp1/ -2 /dev/shm/ -d /mnt/nasext/ -p sumlettersandnumbers -f sumlettersandnumbers > test.out 2>&1 &
[1] 3223
(venv) chia@wvm-chia-02:~/chia-plotter/build$ tail -f test.out
Final Directory: /mnt/nasext/
Number of Plots: 1
Process ID: 3223
Number of Threads: 16
Number of Buckets: 2^7 (128)
Pool Public Key:   nope
Farmer Public Key: ah-no
Working Directory:   /mnt/tmp1/
Working Directory 2: /dev/shm/
Plot Name: plot-k32-2021-06-09-21-02-842af66XXX
[P1] Table 1 took 22.5526 sec
[P1] Table 2 took 161.948 sec, found 4294968849 matches
[P1] Table 3 took 194.87 sec, found 4294951371 matches
[P1] Table 4 took 271.967 sec, found 4295083333 matches
[P1] Table 5 took 254.252 sec, found 4295086160 matches
[P1] Table 6 took 251.612 sec, found 4295093871 matches
[P1] Table 7 took 180.158 sec, found 4295021791 matches
Phase 1 took 1337.4 sec
[P2] max_table_size = 4295093871
[P2] Table 7 scan took 13.3038 sec
[P2] Table 7 rewrite took 39.7664 sec, dropped 0 entries (0 %)

Anyway, thanks a ton for this!

**Comment by F1DNA** [comment]
The final results were, took longer to transfer to the nfs mount over 1GBe than it did to create the plot.  But overall, this is very impressive work.  So much for Chia devs thinking k32 plots will be good for 10 years.  If an R520 and R720 are able to put out a plot in 45-50 minutes with the hardware we have, I'm sure we won't see k32 plots being viable for 10 years.

[P1] Table 1 took 22.5526 sec
[P1] Table 2 took 161.948 sec, found 4294968849 matches
[P1] Table 3 took 194.87 sec, found 4294951371 matches
[P1] Table 4 took 271.967 sec, found 4295083333 matches
[P1] Table 5 took 254.252 sec, found 4295086160 matches
[P1] Table 6 took 251.612 sec, found 4295093871 matches
[P1] Table 7 took 180.158 sec, found 4295021791 matches
Phase 1 took 1337.4 sec
[P2] max_table_size = 4295093871
[P2] Table 7 scan took 13.3038 sec
[P2] Table 7 rewrite took 39.7664 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 44.1593 sec
[P2] Table 6 rewrite took 78.9161 sec, dropped 581332156 entries (13.5348 %)
[P2] Table 5 scan took 52.7882 sec
[P2] Table 5 rewrite took 84.9217 sec, dropped 762038916 entries (17.7421 %)
[P2] Table 4 scan took 51.2604 sec
[P2] Table 4 rewrite took 74.1399 sec, dropped 828934585 entries (19.2996 %)
[P2] Table 3 scan took 48.0081 sec
[P2] Table 3 rewrite took 72.2806 sec, dropped 855043150 entries (19.9081 %)
[P2] Table 2 scan took 56.3407 sec
[P2] Table 2 rewrite took 75.3693 sec, dropped 865559284 entries (20.1529 %)
Phase 2 took 713.865 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 67.275 sec, wrote 3429409565 right entries
[P3-2] Table 2 took 56.1718 sec, wrote 3429409565 left entries, 3429409565 final
[P3-1] Table 3 took 103.016 sec, wrote 3439908221 right entries
[P3-2] Table 3 took 55.4699 sec, wrote 3439908221 left entries, 3439908221 final
[P3-1] Table 4 took 97.5917 sec, wrote 3466148748 right entries
[P3-2] Table 4 took 57.04 sec, wrote 3466148748 left entries, 3466148748 final
[P3-1] Table 5 took 111.91 sec, wrote 3533047244 right entries
[P3-2] Table 5 took 58.0215 sec, wrote 3533047244 left entries, 3533047244 final
[P3-1] Table 6 took 113.544 sec, wrote 3713761715 right entries
[P3-2] Table 6 took 64.7142 sec, wrote 3713761715 left entries, 3713761715 final
[P3-1] Table 7 took 69.9979 sec, wrote 4295021791 right entries
[P3-2] Table 7 took 70.5531 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 931.771 sec, wrote 21877242789 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 141.084 sec, final plot size is 108835896411 bytes
Total plot creation time was 3124.2 sec
Started copy to /mnt/nasext/plot-k32-2021-06-09-21-02-842af6XXX3.plot
Copy to /mnt/nasext/plot-k32-2021-06-09-21-02-842af66d3XXX3.plot finished, took 4019.77 sec, 25.8209 MB/s avg.

**Reply by aaron-morph** [reply]
Thanks for showing an example run, this was really helpful for my first test setup 👍 

I am also running on an old R720

**Reply by F1DNA** [reply]
NP, I have more.  Im running this plotter on a single SATA SSD with a 4th gen i5, another 4thgen i5 on 2 sata SSDs - no raid and then another setup on an old powerede t110 with spinning disks in RAID.  So far, it is faster in every scenario.

**Reply by F1DNA** [reply]
Plus there is this and I had some downtime today as I changed some things around:

![image](https://user-images.githubusercontent.com/61926834/121622154-1d617080-ca33-11eb-8008-ea28c819b4ac.png)

**Reply by F1DNA** [reply]
From 12 plots a day to 22.  Tomorrow will be even better.

**Reply by aaron-morph** [reply]
Noice, my last plot came in at 2882 seconds, not bad for 8 year old Xeons running on 10600 ram

**Reply by aaron-morph** [reply]
@F1DNA , I copied/wrote this little script schedule my plot script which is basically a one liner to plot

#! /bin/sh
# runs for 10 hours
runtime="600 minute"
# ends after 10 hours
endtime=$(date -ud "$runtime" +%s)
while [[ $(date -u +%s) -le $endtime ]]
do
    # run my plot
    ./plotnow.sh
   # sleeps for 4500 seconds before kicking off next plot
    sleep 4500
done

**Reply by kentgran** [reply]
But why do you want it to sleep?

**Reply by aaron-morph** [reply]
yeah that was my bad, I actually brought that down to 60s, probably no need for it but just want a safe pause in between runs

**Reply by kentgran** [reply]
I think it's probably better to run chia-plotter withour any managing scripts, as you will miss a lot of time by not using the built-in staggering for file-copying. I would simply use the "-n -1" instead. Better to spend some scripting forces into analyzing logs or a watchdog monitoring if the plotter is still running. Just a thought.

**Reply by F1DNA** [reply]
This plotter sleeps if you have 2 finished plots that have not been fully copied to your destination yet.  No need to sleep at all.  Now to fix my bottleneck of using an NFS share over a 1GBe connection shared with other resources.

**Reply by aaron-morph** [reply]
good note on the sleep, thanks

@kentgran I haven't used all fo the builtin commands yet but I am planning on being away from my plotter for a few days this weekend so just wanted to make sure it was running with a script, while I was away, lol

**Reply by aaron-morph** [reply]
@F1DNA so now I see the --count, is that serial and not parallel, so I could just set it for 12 plots and it would run them one after the other?

**Reply by F1DNA** [reply]
The current documentation gives options of 1 or -1(infinite).  I have not tested to see if you can pass any other value.  Likely you can.  So if you know your destination has room for 8 more plots, I would think you can specify that.  Test it out by tossing it a "2" and see what happens.

**Comment by kentgran** [comment]
I've been using a Corsair MP600 PRO 1TB as tmp and a normal Corsair MP600 500GB as tmp2 on a computer with a Ryzen 7 / 64GB running Plotman. This has outputted a k23 every 1h20m or so. It could probably have been tuned better, but I decided not to waste more time on that.

Now, running on the same computer, I'm only using the faster m.2 (PRO), which obviously only now gets utilized to its full potential. I get another k23 every 40 minutes!

Another problem solved: Now, the factory can be simply stopped, recompiled and started again - without losing too much. Whereas before, running on this staggered monster, it would take more than 6 hours stopping it. And the lost production would still be enormous as the staggering fades out very slowly.

Thank you for making it easier!

I will now only have to sort out how to move the plots to the harvester fast enough that the plotter doesn't get full, then I will send you some chiatoshi as soon as I get in contact with my, until now, cold wallet.

**Reply by F1DNA** [reply]
k23?  I hope you mean k32...

**Reply by kentgran** [reply]
Sure! I meant k32. And I was not exaggerating, the files are just 36  and 37 minutes between each. That's more than triple my old speed.

**Reply by onurcanbjk** [reply]
Can u share your configs? Hw many thread using and others?
 I have Ryzen 5 5600x 32 gb ram and 2 tb mp 600 corsair, using 18 thread plot time is 2 hours.

**Reply by kentgran** [reply]
Ryzen 5 has only 6 cores, I think 12 threads is the best it can do.

`./chia_plot -n -1 -r 16 -p [mypoolkey] -f [myfarmerkey] -t /chiatmp/ -d /chiaplots/ | tee output.log`

**Reply by onurcanbjk** [reply]
Ty for help, this command gave 2 hours. Which is very fast but cant parallel 🙂

**Reply by kentgran** [reply]
I have my 1TB Corsair PRO mounted on /chiatmp and a 7200rpm HDD on /chiaplots in Ubuntu 20.04 LTS. I think your setup should do almost as good as mine (which is down to 36 minutes each plot if I don't do anything else on the computer)

**Reply by T5-R** [reply]
Thanks @ madMAx43v3r for this supercharger!

My results so far, using a Ryzen 5 3600, 32gb ram, 12 threads:
- using singe hdds: ~9hrs per plot
- using nvme temp1, 4*2.5" sata drive Raid0 for temp2: 2hrs
- using 2 512gb nvme (temp1 is a slower 2lane PCIE only): **54 minutes** 👍
